src/opendr/perception/multimodal_human_centric/intent_recognition_learner/intent_recognition_learner.py

```diff
@@ -0,0 +1,289 @@
+        If False, current model state will be evaluated.
+        :type restore_best_model: bool
+        :return: Performance metrics on test set, predicted labels.
+        :rtype: dict
+        """
+        assert modality in ['audio', 'video', 'language', 'joint'], 'Unknown modality: {}'.format(modality)
+        assert (
+            modality == self.train_config.mode or self.train_config.mode == 'joint'), \
+            'Inference on modality {} not supported with mode {}'.format(
+            modality, self.train_config.mode)
+
+        self.__update_verbosity(silent, verbose)
+
+        dataloader = DataLoader(dataset, batch_size=self.train_config.test_batch_size,
+                                num_workers=self.train_config.num_workers, pin_memory=True)
+
+        self.logger.info("Started testing...")
+        outputs = self.method.test(dataloader, modality, restore_best_model)
+
+        save_results(self.train_config, outputs, suff=modality, results_file_name=modality + '_results.csv')
+
+        self.logger.info("Finished testing...")
+
+        return outputs
+
+    def infer(self, batch, modality='language'):
+        """
+        Splits the input text into sentences, process each sentence independently.
+        If a sentence is > max_sequence_length, process in sliding window manner.
+        :parameter batch: Input data
+        :type batch: dict
+        :parameter modality: Specifiec inference modality
+        :type modality: str
+        :return: Predicted class label with confidence score for each sentence in the input
+        :rtype: list of opendr.data.Category
+        """
+        assert modality in ['language'], 'Unsupported modality {}'.format(modality)
+        if self.sentence_tokenizer is None:
+            self.sentence_tokenizer = nltk.data.load('tokenizers/punkt/english.pickle')
+
+        out = []
+        for sentence in self.sentence_tokenizer.tokenize(batch['text']):
+            text_feats = self.__process_raw_text(sentence)
+            pred = self.method.infer(text_feats, modality=modality)
+            out.append(Category(int(pred[0]), confidence=pred[1]))
+        return out
+
+    def save(self, path):
+        """ Saves current state of the model to the given path
+
+        :parameter path: Path where to save the model
+        :type path: str
+        """
+        self.logger.info("Saving model...")
+        state_dict = {
+            'state_dict': self.model.model.state_dict(),
+            'config': self.train_config
+        }
+        torch.save(state_dict, path)
+        self.logger.info("Save model.")
+
+    def load(self, path):
+        """ Loads model chekpoint from specified path
+
+        :parameter path: Path to the checkpoint
+        :type path: str
+        """
+        self.logger.info("Loading model from {}...".format(path))
+        checkpoint = torch.load(path, map_location=self.train_config.device)
+        self.model.model.load_state_dict(checkpoint['state_dict'])
```

Comment:
Thank you for looking into this! Looking deeper, it seems this inconsistency arises from how different versions of huggingface treat position_ids during creation/model saving and I had been using an older version when creating the checkpoints. That being said, I believe setting strict=False is fine as a resolution since in any case we are using the default ids. I have checked that the model seems to be behaving as it sholuld.

Reply:
Thank you very much!